voyager_donuts.py

```diff
@@ -1171,12 +1171,15 @@ def __receive_until_delim(self, delim=b'\r\n'):
 
         # check if there is any overflow from last time
         logging.debug(f"Message overflow {self.message_overflow}")
-        for msg in self.message_overflow:
-            logging.debug(f"Moving {msg} to __receive_until_delim message_buffer...")
-            message_buffer.append(msg)
+
+        if self.message_overflow:
+            logging.debug(f"Moving {self.message_overflow} to __receive_until_delim initial_message_raw...")
+            initial_message_raw = self.message_overflow
+        else:
+            initial_message_raw = b''
```

Comment:
The issue with ``message_overflow`` as a list is that it may end up containing multiple separate JSON chunks which should not all be added to the buffer. With the approach here, we don't split/modify the unused part of the message, and we just prepend it to the raw data.